tests/core/rmdir.c

```diff
@@ -27,10 +27,24 @@ void test_core_rmdir__initialize(void)
 	git_buf_dispose(&path);
 }
 
+void test_core_rmdir__cleanup(void)
+{
+	if (git_path_exists(empty_tmp_dir))
+		cl_git_pass(git_futils_rmdir_r(empty_tmp_dir, NULL, GIT_RMDIR_REMOVE_FILES));
```

Comment:
Not if a test has been written that deletes the directory itself

Reply:
But that's not clar's lifecycle - for every test, the `initialize` will run immediately before, and the `cleanup` will run immediately after.  So `core::rmdir::can_skip_non_empty_dir` can do whatever it wants, the `initialize` will recreate that directory.

If that directory doesn't exist, then something strange has happened in the test execution, and we should error so that we can investigate, IMHO.

Reply:
The current flow is:

1. The `__initialize` function creates `empty_tmp_dir` and some more directories inside of it
2. The test performs whatever it wants with and within empty_tmp_dir. At least one test will use `git_futils_rmdir_r` to remove `empty_tmp_dir` altogether
3. The `__cleanup` function will now try to remove `empty_tmp_dir` again. But as the test has already deleted it, it will fail

So it's wasn't anything strange, but just the test verifying that it is able to remove `empty_tmp_dir` with the function that is currently the system under test

Reply:
Oh, goodness.  I'm very sorry, I thought that this was another _test_ being added, I hadn't realized that it was the cleanup function itself.  😥  Thanks for reminding me.